core/src/channel/voice_buffer.rs

```diff
@@ -107,20 +111,44 @@ impl VoiceBuffer {
 
         // Find the first non releasing voice, get its id and release all voices with that id
         for voice in self.buffer.iter_mut() {
-            if voice.is_releasing() {
-                continue;
-            }
-
-            if id.is_none() {
-                id = Some(voice.id);
-                vel = Some(voice.velocity())
-            }
-
-            if id != Some(voice.id) {
-                break;
+            if self.damper {
+                self.held_by_damper.push(voice.id.clone());
+            } else {
+                if voice.is_releasing() {
+                    continue;
+                }
+
+                if id.is_none() {
+                    id = Some(voice.id);
+                    vel = Some(voice.velocity())
+                }
+
+                if id != Some(voice.id) {
+                    break;
+                }
+
+                voice.signal_release();
+
+                for v in &mut self.held_by_damper {
```

Comment:
Could replace this with `for v in self.held_by_damper.drain(..)` because that will automatically "drain" all the items in the array, which means you don't have to use .clear later

Reply:
Oh interesting, I didn't know about this function. I'll change it.